### PR TITLE
Sequence composition

### PR DESCRIPTION
**Pull request by edawson**
Will hold functions for composing sequences from FASTA + VCF (before generating kmer indexes).

### CODE REVIEW

**Comment by pb-cdunn** [comment]
Looking good.

File: niblib/compose.nim [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import hts
+
+type
+  FlankSeq = object
+    left, right: string
+
+proc retrieve_flanking_sequences_from_fai(fastaIdx: Fai, chrom: string,
+        start_pos, end_pos, flank: int) : FlankSeq  =
+    ## this function lacks a return
+    result.left = fastaIdx.get(chrom, start_pos - flank, start_pos)
```
Comment (brentp) [inline niblib/compose.nim]: `max(0, start_pos - flank)`

after adding `import math`.

File: niblib/compose.nim [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  var sv_end = new_seq[int32](2)
+  var sv_start: int32
+  var sv_type: string
+  for v in variants:
+    var info_fields = v.info
+    ## Extract SV type / SV END / start position
+    doAssert info_fields.get("SVTYPE", sv_type) == Status.OK
+    doAssert info_fields.get("END", sv_end) == Status.OK
+    let sv_start = v.POS
```
Comment (brentp) [inline niblib/compose.nim:35]: v.POS is 1-based, use v.start

File: niblib/compose.nim [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  doAssert(open(variants, variant_file))
+
+  var sv_end = new_seq[int32](2)
+  var sv_start: int32
+  var sv_type: string
+  for v in variants:
+    var info_fields = v.info
+    ## Extract SV type / SV END / start position
+    doAssert info_fields.get("SVTYPE", sv_type) == Status.OK
+    doAssert info_fields.get("END", sv_end) == Status.OK
```
Comment (brentp) [inline niblib/compose.nim:34]: sv_end = v.stop

**Pull request merged**